internal/metamorphic/crossversion/crossversion_test: TestMetaCrossVersion failed

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14483157?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14483157?buildTab=artifacts#/) on refs/heads/master @ [8a097e8ae86b](https://github.com/cockroachdb/pebble/commits/8a097e8ae86b):


```
                --- PASS: TestMeta/execution/standard-021 (5.19s)
                --- FAIL: TestMeta/execution/random-019 (0.53s)
                --- PASS: TestMeta/execution/standard-027 (5.33s)
                --- PASS: TestMeta/execution/random-000 (5.69s)
                --- PASS: TestMeta/execution/random-020 (2.90s)
                --- PASS: TestMeta/execution/random-017 (2.30s)
                --- PASS: TestMeta/execution/random-018 (2.36s)
                --- PASS: TestMeta/execution/random-015 (2.90s)
                --- PASS: TestMeta/execution/random-016 (3.55s)
                --- FAIL: TestMeta/execution/random-011 (0.68s)
                --- PASS: TestMeta/execution/random-014 (2.60s)
                --- PASS: TestMeta/execution/random-012 (2.11s)
                --- PASS: TestMeta/execution/random-027 (7.20s)
                --- FAIL: TestMeta/execution/random-009 (0.84s)
                --- FAIL: TestMeta/execution/random-006 (0.83s)
                --- PASS: TestMeta/execution/random-013 (4.19s)
                --- FAIL: TestMeta/execution/random-004 (1.57s)
                --- PASS: TestMeta/execution/random-008 (3.20s)
                --- FAIL: TestMeta/execution/random-002 (0.93s)
                --- PASS: TestMeta/execution/random-007 (3.89s)
                --- PASS: TestMeta/execution/random-005 (4.31s)
                --- PASS: TestMeta/execution/random-003 (3.06s)
                --- PASS: TestMeta/execution/random-001 (2.86s)
                --- PASS: TestMeta/execution/standard-023 (2.70s)
                --- PASS: TestMeta/execution/standard-024 (3.49s)
                --- PASS: TestMeta/execution/standard-022 (3.16s)
                --- PASS: TestMeta/execution/standard-017 (2.90s)
                --- PASS: TestMeta/execution/random-010 (9.29s)
                --- PASS: TestMeta/execution/standard-019 (2.62s)
                --- PASS: TestMeta/execution/standard-018 (2.92s)
                --- PASS: TestMeta/execution/standard-007 (2.42s)
                --- PASS: TestMeta/execution/standard-016 (3.61s)
                --- PASS: TestMeta/execution/standard-012 (2.38s)
                --- PASS: TestMeta/execution/standard-013 (3.18s)
                --- PASS: TestMeta/execution/standard-004 (2.32s)
                --- PASS: TestMeta/execution/standard-006 (2.51s)
                --- PASS: TestMeta/execution/random-025 (19.64s)
                --- PASS: TestMeta/execution/standard-009 (3.09s)
                --- PASS: TestMeta/execution/standard-005 (3.89s)
                --- PASS: TestMeta/execution/standard-010 (3.17s)
                --- PASS: TestMeta/execution/standard-002 (3.36s)
                --- PASS: TestMeta/execution/standard-003 (2.42s)
                --- PASS: TestMeta/execution/standard-001 (2.77s)
                --- PASS: TestMeta/execution/standard-011 (1.60s)
                --- PASS: TestMeta/execution/standard-015 (8.56s)
                --- PASS: TestMeta/execution/standard-008 (2.62s)
                --- PASS: TestMeta/execution/random-023 (23.55s)
        FAIL
        
        --- FAIL: TestMetaCrossVersion/execution-crl-release-23.2/sha=78ab2ba8-seed=10117690363656650527-opts=random-002(sha=1becf831-seed=7907179883047621387-opts=standard-020(sha=ffd5ce9c-seed=6469414836748746302-opts=standard-017(<empty>))) (206.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14501938?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14501938?buildTab=artifacts#/) on refs/heads/master @ [e5c9f635e505](https://github.com/cockroachdb/pebble/commits/e5c9f635e505):


```
                --- FAIL: TestMeta/execution/random-024 (0.85s)
                --- PASS: TestMeta/execution/random-012 (8.20s)
                --- FAIL: TestMeta/execution/random-003 (0.99s)
                --- PASS: TestMeta/execution/random-023 (3.65s)
                --- PASS: TestMeta/execution/random-005 (11.16s)
                --- PASS: TestMeta/execution/random-020 (5.08s)
                --- FAIL: TestMeta/execution/random-018 (1.02s)
                --- PASS: TestMeta/execution/random-019 (6.40s)
                --- PASS: TestMeta/execution/random-004 (4.72s)
                --- PASS: TestMeta/execution/random-025 (8.85s)
                --- PASS: TestMeta/execution/random-017 (6.06s)
                --- PASS: TestMeta/execution/random-016 (5.82s)
                --- PASS: TestMeta/execution/random-008 (19.50s)
                --- PASS: TestMeta/execution/standard-026 (3.78s)
                --- PASS: TestMeta/execution/random-000 (7.01s)
                --- PASS: TestMeta/execution/standard-027 (6.55s)
                --- PASS: TestMeta/execution/standard-025 (4.23s)
                --- PASS: TestMeta/execution/standard-023 (4.90s)
                --- PASS: TestMeta/execution/standard-022 (4.49s)
                --- PASS: TestMeta/execution/standard-024 (6.47s)
                --- PASS: TestMeta/execution/standard-019 (2.21s)
                --- PASS: TestMeta/execution/standard-020 (5.81s)
                --- PASS: TestMeta/execution/standard-015 (16.51s)
                --- PASS: TestMeta/execution/standard-018 (4.97s)
                --- PASS: TestMeta/execution/standard-017 (4.43s)
                --- PASS: TestMeta/execution/standard-021 (9.23s)
                --- PASS: TestMeta/execution/random-010 (3.82s)
                --- PASS: TestMeta/execution/standard-016 (5.01s)
                --- PASS: TestMeta/execution/random-002 (20.93s)
                --- PASS: TestMeta/execution/standard-014 (5.21s)
                --- PASS: TestMeta/execution/standard-012 (5.28s)
                --- PASS: TestMeta/execution/standard-013 (5.45s)
                --- PASS: TestMeta/execution/standard-008 (7.37s)
                --- PASS: TestMeta/execution/standard-011 (5.13s)
                --- PASS: TestMeta/execution/random-013 (7.74s)
                --- PASS: TestMeta/execution/standard-010 (4.43s)
                --- PASS: TestMeta/execution/random-021 (34.86s)
                --- PASS: TestMeta/execution/standard-009 (4.11s)
                --- PASS: TestMeta/execution/standard-004 (4.50s)
                --- PASS: TestMeta/execution/random-011 (12.87s)
                --- PASS: TestMeta/execution/standard-007 (4.43s)
                --- PASS: TestMeta/execution/standard-006 (4.56s)
                --- PASS: TestMeta/execution/random-009 (5.34s)
                --- PASS: TestMeta/execution/standard-002 (2.34s)
                --- PASS: TestMeta/execution/standard-005 (3.61s)
                --- PASS: TestMeta/execution/standard-003 (2.37s)
                --- PASS: TestMeta/execution/standard-001 (1.73s)
        FAIL
        
        --- FAIL: TestMetaCrossVersion/execution-crl-release-23.2/sha=78ab2ba8-seed=5542476348556447836-opts=standard-010(sha=1becf831-seed=4663534632829966196-opts=random-002(sha=ffd5ce9c-seed=16547645599882799100-opts=random-008(<empty>))) (340.59s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14521526?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14521526?buildTab=artifacts#/) on refs/heads/master @ [bd8563ecc1ad](https://github.com/cockroachdb/pebble/commits/bd8563ecc1ad):


```
                --- PASS: TestMeta/execution/random-025 (3.17s)
                --- PASS: TestMeta/execution/random-023 (2.78s)
                --- PASS: TestMeta/execution/random-020 (2.28s)
                --- PASS: TestMeta/execution/random-021 (2.81s)
                --- FAIL: TestMeta/execution/random-016 (0.90s)
                --- PASS: TestMeta/execution/random-018 (2.89s)
                --- PASS: TestMeta/execution/random-015 (2.76s)
                --- PASS: TestMeta/execution/standard-004 (2.58s)
                --- PASS: TestMeta/execution/random-017 (5.28s)
                --- PASS: TestMeta/execution/standard-006 (3.02s)
                --- PASS: TestMeta/execution/random-014 (11.73s)
                --- PASS: TestMeta/execution/standard-002 (2.33s)
                --- PASS: TestMeta/execution/standard-005 (3.95s)
                --- PASS: TestMeta/execution/standard-003 (2.68s)
                --- FAIL: TestMeta/execution/random-009 (0.46s)
                --- PASS: TestMeta/execution/random-008 (1.51s)
                --- PASS: TestMeta/execution/random-011 (3.70s)
                --- PASS: TestMeta/execution/random-010 (2.71s)
                --- PASS: TestMeta/execution/random-019 (11.14s)
                --- PASS: TestMeta/execution/random-012 (4.40s)
                --- PASS: TestMeta/execution/random-007 (6.90s)
                --- PASS: TestMeta/execution/standard-026 (2.57s)
                --- PASS: TestMeta/execution/standard-024 (2.24s)
                --- PASS: TestMeta/execution/standard-023 (2.25s)
                --- PASS: TestMeta/execution/standard-025 (3.46s)
                --- PASS: TestMeta/execution/standard-022 (1.79s)
                --- PASS: TestMeta/execution/standard-010 (2.97s)
                --- PASS: TestMeta/execution/standard-011 (2.83s)
                --- PASS: TestMeta/execution/standard-008 (1.91s)
                --- PASS: TestMeta/execution/standard-009 (2.50s)
                --- PASS: TestMeta/execution/standard-012 (4.24s)
                --- PASS: TestMeta/execution/standard-021 (7.78s)
                --- PASS: TestMeta/execution/standard-001 (2.45s)
                --- PASS: TestMeta/execution/random-006 (1.96s)
                --- PASS: TestMeta/execution/random-026 (21.83s)
                --- PASS: TestMeta/execution/standard-018 (1.76s)
                --- PASS: TestMeta/execution/random-005 (3.80s)
                --- PASS: TestMeta/execution/random-001 (1.97s)
                --- PASS: TestMeta/execution/random-002 (3.48s)
                --- PASS: TestMeta/execution/standard-019 (0.93s)
                --- PASS: TestMeta/execution/standard-020 (2.61s)
                --- PASS: TestMeta/execution/standard-017 (1.83s)
                --- PASS: TestMeta/execution/standard-016 (1.96s)
                --- PASS: TestMeta/execution/random-027 (26.14s)
                --- FAIL: TestMeta/execution/random-004 (6.87s)
                --- PASS: TestMeta/execution/random-003 (7.17s)
                --- PASS: TestMeta/execution/standard-015 (4.17s)
        FAIL
        
        --- FAIL: TestMetaCrossVersion/execution-crl-release-23.2/sha=78ab2ba8-seed=279439816562517169-opts=standard-015(sha=1becf831-seed=1078422020815320700-opts=standard-023(sha=ffd5ce9c-seed=10179031060247823564-opts=random-009(<empty>))) (227.53s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14532395?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14532395?buildTab=artifacts#/) on refs/heads/master @ [86b5f54ca026](https://github.com/cockroachdb/pebble/commits/86b5f54ca026):


```
                --- PASS: TestMeta/execution/standard-016 (1.23s)
                --- PASS: TestMeta/execution/standard-006 (1.09s)
                --- PASS: TestMeta/execution/standard-017 (1.83s)
                --- PASS: TestMeta/execution/standard-014 (1.55s)
                --- PASS: TestMeta/execution/standard-012 (1.22s)
                --- PASS: TestMeta/execution/standard-015 (1.88s)
                --- PASS: TestMeta/execution/random-001 (4.52s)
                --- PASS: TestMeta/execution/standard-021 (2.85s)
                --- PASS: TestMeta/execution/standard-011 (0.85s)
                --- PASS: TestMeta/execution/standard-009 (0.80s)
                --- PASS: TestMeta/execution/standard-003 (0.47s)
                --- PASS: TestMeta/execution/standard-007 (0.58s)
                --- PASS: TestMeta/execution/standard-010 (1.18s)
                --- PASS: TestMeta/execution/standard-005 (0.60s)
                --- PASS: TestMeta/execution/standard-008 (1.08s)
                --- PASS: TestMeta/execution/standard-002 (0.63s)
                --- PASS: TestMeta/execution/standard-004 (0.69s)
                --- PASS: TestMeta/execution/standard-001 (0.70s)
        FAIL
        
        --- FAIL: TestMetaCrossVersion/execution-crl-release-23.2/sha=78ab2ba8-seed=5567011436274258280-opts=standard-003(sha=1becf831-seed=15557802357884708649-opts=random-021(sha=ffd5ce9c-seed=14564169981444554143-opts=random-021(<empty>))) (134.17s)
=== RUN   TestMetaCrossVersion/execution-crl-release-23.2
    --- FAIL: TestMetaCrossVersion/execution-crl-release-23.2 (0.00s)
=== RUN   TestMetaCrossVersion/execution-crl-release-23.2/sha=78ab2ba8-seed=5567011436274258280-opts=random-014(sha=1becf831-seed=15557802357884708649-opts=standard-020(sha=ffd5ce9c-seed=14564169981444554143-opts=random-011(<empty>)))
=== PAUSE TestMetaCrossVersion/execution-crl-release-23.2/sha=78ab2ba8-seed=5567011436274258280-opts=random-014(sha=1becf831-seed=15557802357884708649-opts=standard-020(sha=ffd5ce9c-seed=14564169981444554143-opts=random-011(<empty>)))
=== CONT  TestMetaCrossVersion/execution-crl-release-23.2/sha=78ab2ba8-seed=5567011436274258280-opts=random-014(sha=1becf831-seed=15557802357884708649-opts=standard-020(sha=ffd5ce9c-seed=14564169981444554143-opts=random-011(<empty>)))
    crossversion_test.go:248:   Running test with version v1.1.0 with initial state sha=78ab2ba8-seed=5567011436274258280-opts=random-014(sha=1becf831-seed=15557802357884708649-opts=standard-020(sha=ffd5ce9c-seed=14564169981444554143-opts=random-011(<empty>))).
        --- PASS: TestMetaCrossVersion/execution-crl-release-23.2/sha=78ab2ba8-seed=5567011436274258280-opts=random-014(sha=1becf831-seed=15557802357884708649-opts=standard-020(sha=ffd5ce9c-seed=14564169981444554143-opts=random-011(<empty>))) (3.13s)
panic: runtime error: index out of range [142] with length 142 [recovered]
	panic: runtime error: index out of range [142] with length 142

goroutine 9 [running]:
testing.tRunner.func1.2({0xda7ac0, 0xc000044288})
	GOROOT/src/testing/testing.go:1631 +0x24a
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1634 +0x377
panic({0xda7ac0?, 0xc000044288?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/pebble/metamorphic.reorderHistory({0xc0002e1508, 0x8e, 0xc00081e750?})
	github.com/cockroachdb/pebble/metamorphic/external/com_github_cockroachdb_pebble/metamorphic/history.go:209 +0x105
github.com/cockroachdb/pebble/metamorphic.CompareHistories({0xf22540, 0xc0000d36c0}, {0xc000072f08, 0x1f8, 0x7fff97a2b806?})
	github.com/cockroachdb/pebble/metamorphic/external/com_github_cockroachdb_pebble/metamorphic/history.go:184 +0x105
internal/metamorphic/crossversion/crossversion_test.runCrossVersion({0xf250a0, 0x14d0180}, 0xc0000d36c0, {0xc00004a600, 0x57}, {0xc000002600, 0x5, 0x603?}, 0x17bf97fa5c4ac8ab, 0xa)
	internal/metamorphic/crossversion/crossversion_test/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test.go:190 +0x454
internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion(0xc0000d36c0)
	internal/metamorphic/crossversion/crossversion_test/external/com_github_cockroachdb_pebble/internal/metamorphic/crossversion/crossversion_test.go:124 +0x554
testing.tRunner(0xc0000d36c0, 0xe51828)
	GOROOT/src/testing/testing.go:1689 +0xfb
created by testing.(*T).Run in goroutine 1
	GOROOT/src/testing/testing.go:1742 +0x390
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14549256?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14549256?buildTab=artifacts#/) on refs/heads/master @ [86b5f54ca026](https://github.com/cockroachdb/pebble/commits/86b5f54ca026):


```
                --- FAIL: TestMeta/execution/random-021 (1.38s)
                --- PASS: TestMeta/execution/standard-021 (4.14s)
                --- FAIL: TestMeta/execution/random-018 (0.80s)
                --- FAIL: TestMeta/execution/random-016 (0.53s)
                --- PASS: TestMeta/execution/random-024 (2.79s)
                --- PASS: TestMeta/execution/random-000 (5.34s)
                --- PASS: TestMeta/execution/random-025 (4.23s)
                --- PASS: TestMeta/execution/random-015 (2.84s)
                --- PASS: TestMeta/execution/random-014 (3.12s)
                --- PASS: TestMeta/execution/random-013 (3.56s)
                --- FAIL: TestMeta/execution/random-010 (1.01s)
                --- FAIL: TestMeta/execution/random-008 (0.36s)
                --- PASS: TestMeta/execution/random-017 (5.28s)
                --- FAIL: TestMeta/execution/random-009 (1.43s)
                --- PASS: TestMeta/execution/random-005 (2.23s)
                --- PASS: TestMeta/execution/random-007 (3.63s)
                --- PASS: TestMeta/execution/random-006 (3.29s)
                --- FAIL: TestMeta/execution/random-001 (1.36s)
                --- PASS: TestMeta/execution/random-003 (3.15s)
                --- PASS: TestMeta/execution/standard-026 (3.13s)
                --- PASS: TestMeta/execution/random-022 (17.08s)
                --- PASS: TestMeta/execution/standard-027 (6.08s)
                --- PASS: TestMeta/execution/standard-025 (2.39s)
                --- PASS: TestMeta/execution/standard-023 (2.24s)
                --- PASS: TestMeta/execution/standard-024 (2.68s)
                --- PASS: TestMeta/execution/standard-022 (2.22s)
                --- PASS: TestMeta/execution/standard-013 (1.92s)
                --- PASS: TestMeta/execution/standard-001 (2.73s)
                --- PASS: TestMeta/execution/standard-012 (2.77s)
                --- PASS: TestMeta/execution/random-011 (19.39s)
                --- PASS: TestMeta/execution/standard-011 (1.62s)
                --- PASS: TestMeta/execution/random-004 (16.29s)
                --- PASS: TestMeta/execution/random-012 (20.60s)
                --- PASS: TestMeta/execution/standard-010 (2.00s)
                --- PASS: TestMeta/execution/standard-020 (1.35s)
                --- PASS: TestMeta/execution/standard-009 (1.96s)
                --- PASS: TestMeta/execution/standard-018 (1.70s)
                --- PASS: TestMeta/execution/standard-008 (1.91s)
                --- PASS: TestMeta/execution/standard-019 (1.44s)
                --- PASS: TestMeta/execution/standard-016 (1.39s)
                --- PASS: TestMeta/execution/standard-003 (1.13s)
                --- PASS: TestMeta/execution/standard-006 (1.07s)
                --- PASS: TestMeta/execution/standard-017 (1.67s)
                --- PASS: TestMeta/execution/standard-005 (1.18s)
                --- PASS: TestMeta/execution/random-020 (26.96s)
                --- PASS: TestMeta/execution/random-002 (18.11s)
                --- PASS: TestMeta/execution/standard-015 (3.04s)
        FAIL
        
        --- FAIL: TestMetaCrossVersion/execution-crl-release-23.2/sha=78ab2ba8-seed=10881802203431543675-opts=random-023(sha=1becf831-seed=13977821285290097806-opts=standard-024(sha=ffd5ce9c-seed=12507384275020755193-opts=random-021(<empty>))) (212.93s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14559174?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/14559174?buildTab=artifacts#/) on refs/heads/master @ [9c2da6d20e11](https://github.com/cockroachdb/pebble/commits/9c2da6d20e11):


```
                --- PASS: TestMeta/execution/random-007 (0.59s)
                --- PASS: TestMeta/execution/standard-010 (0.85s)
                --- FAIL: TestMeta/execution/random-011 (0.44s)
                --- PASS: TestMeta/execution/standard-008 (0.99s)
                --- PASS: TestMeta/execution/random-012 (0.77s)
                --- PASS: TestMeta/execution/random-013 (1.12s)
                --- PASS: TestMeta/execution/random-009 (0.81s)
                --- PASS: TestMeta/execution/random-004 (0.79s)
                --- PASS: TestMeta/execution/random-006 (0.90s)
                --- PASS: TestMeta/execution/random-010 (1.39s)
                --- PASS: TestMeta/execution/random-001 (0.79s)
                --- FAIL: TestMeta/execution/random-008 (1.64s)
                --- PASS: TestMeta/execution/random-021 (0.79s)
                --- PASS: TestMeta/execution/random-003 (1.32s)
                --- FAIL: TestMeta/execution/random-027 (0.73s)
                --- PASS: TestMeta/execution/random-026 (0.80s)
                --- PASS: TestMeta/execution/random-023 (0.77s)
                --- PASS: TestMeta/execution/random-002 (2.09s)
                --- FAIL: TestMeta/execution/random-022 (0.89s)
                --- PASS: TestMeta/execution/standard-003 (0.81s)
                --- PASS: TestMeta/execution/standard-004 (0.85s)
                --- PASS: TestMeta/execution/random-025 (1.75s)
                --- FAIL: TestMeta/execution/random-024 (2.02s)
                --- PASS: TestMeta/execution/standard-026 (0.96s)
                --- PASS: TestMeta/execution/standard-025 (0.92s)
                --- PASS: TestMeta/execution/standard-018 (0.46s)
                --- PASS: TestMeta/execution/standard-024 (1.35s)
                --- PASS: TestMeta/execution/standard-027 (1.80s)
                --- PASS: TestMeta/execution/standard-002 (0.69s)
                --- PASS: TestMeta/execution/standard-023 (1.16s)
                --- PASS: TestMeta/execution/standard-020 (0.47s)
                --- PASS: TestMeta/execution/standard-022 (1.11s)
                --- PASS: TestMeta/execution/standard-019 (0.59s)
                --- PASS: TestMeta/execution/random-018 (0.53s)
                --- FAIL: TestMeta/execution/random-019 (0.38s)
                --- PASS: TestMeta/execution/standard-021 (2.61s)
                --- PASS: TestMeta/execution/standard-001 (0.83s)
                --- PASS: TestMeta/execution/random-016 (0.47s)
                --- PASS: TestMeta/execution/standard-016 (0.71s)
                --- PASS: TestMeta/execution/standard-017 (0.61s)
                --- PASS: TestMeta/execution/random-020 (1.18s)
                --- PASS: TestMeta/execution/random-015 (0.49s)
                --- PASS: TestMeta/execution/random-017 (0.73s)
                --- PASS: TestMeta/execution/standard-015 (0.73s)
                --- PASS: TestMeta/execution/random-005 (5.74s)
        FAIL
        
        --- FAIL: TestMetaCrossVersion/execution-crl-release-23.2/sha=78ab2ba8-seed=18397719996466739288-opts=standard-014(sha=1becf831-seed=5294842004831360724-opts=random-010(sha=ffd5ce9c-seed=11038209210712196020-opts=random-015(<empty>))) (65.22s)
=== RUN   TestMetaCrossVersion/execution-crl-release-23.2
    --- FAIL: TestMetaCrossVersion/execution-crl-release-23.2 (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/metamorphic/options.go
+++ b/metamorphic/options.go
@@ -517,16 +517,6 @@ func randomOptions(
 	default:
 		lopts.FilterPolicy = newTestingFilterPolicy(1 << rng.Intn(5))
 	}
-
-	// We use either no compression, snappy compression or zstd compression.
-	switch rng.Intn(3) {
-	case 0:
-		lopts.Compression = pebble.NoCompression
-	case 1:
-		lopts.Compression = pebble.ZstdCompression
-	default:
-		lopts.Compression = pebble.SnappyCompression
-	}
 	opts.Levels = []pebble.LevelOptions{lopts}
 
 	// Explicitly disable disk-backed FS's for the random configurations. The

